core/internal/clients/retry_policy.go

```diff
@@ -34,9 +35,13 @@ func UpsertBucketRetryPolicy(ctx context.Context, resp *http.Response, err error
 	statusCode := resp.StatusCode
 	switch {
 	case statusCode == http.StatusGone: // don't retry on 410 Gone
-		return false, fmt.Errorf("the server responded with an error. (Error %d: %s)", statusCode, http.StatusText(statusCode))
+		return false, err
 	case statusCode == http.StatusConflict: // retry on 409 Conflict
-		return true, fmt.Errorf("conflict, retrying. (Error %d: %s)", statusCode, http.StatusText(statusCode))
+		return true, err
+	case statusCode == http.StatusBadRequest: // don't retry on 400 bad request
+		return false, err
+	case statusCode == http.StatusUnprocessableEntity:
```

Comment:
we can probably just halt on this one everywhere; we return it very few places and all of them so far should be definite, non-retryable errors as far as i know